src/LeanCode.ContractsGenerator/ContractTypes.cs

```diff
@@ -10,33 +10,52 @@ namespace LeanCode.ContractsGenerator
 {
     public sealed class ContractTypes
     {
-        public INamedTypeSymbol QueryType { get; }
-        public INamedTypeSymbol CommandType { get; }
+        private HashSet<INamedTypeSymbol> QueryType { get; }
+        private HashSet<INamedTypeSymbol> CommandType { get; }
 
-        public INamedTypeSymbol AuthorizeWhenAttribute { get; }
-        public INamedTypeSymbol AuthorizeWhenHasAnyOfAttribute { get; }
-        public INamedTypeSymbol QueryCacheAttribute { get; }
+        private HashSet<INamedTypeSymbol> AuthorizeWhenAttribute { get; }
+        private HashSet<INamedTypeSymbol> AuthorizeWhenHasAnyOfAttribute { get; }
+        private HashSet<INamedTypeSymbol> QueryCacheAttribute { get; }
 
-        public INamedTypeSymbol ExcludeFromContractsGenerationAttribute { get; }
+        private HashSet<INamedTypeSymbol> ExcludeFromContractsGenerationAttribute { get; }
 
-        public INamedTypeSymbol Attribute { get; }
-        public INamedTypeSymbol AttributeUsageAttribute { get; }
-        public INamedTypeSymbol ReadOnlyDictionary { get; }
-        public INamedTypeSymbol Dictionary { get; }
+        private HashSet<INamedTypeSymbol> Attribute { get; }
+        private HashSet<INamedTypeSymbol> AttributeUsageAttribute { get; }
+        private HashSet<INamedTypeSymbol> ReadOnlyDictionary { get; }
+        private HashSet<INamedTypeSymbol> Dictionary { get; }
```

Comment:
I'd use `ImmutableHashSet` and explicitly pass `SymbolEqualityComparer.Default` as its comparer.

Reply:
There is no profit in using `ImmutableHashSet`. Passing `SymbolEqualityComparer.Default`, unfortunately, does not work because of type mismatch (it compares `ISymbol` and the resulting `HashSet` needs to be on `ISymbol` not on `INamedTypeSymbol` :(). Creating a wrapper brings no benefit.